addons/sale_stock/models/sale_order.py

```diff
@@ -425,9 +410,12 @@ def create(self, vals_list):
     def write(self, values):
         lines = self.env['sale.order.line']
         if 'product_uom_qty' in values:
-            precision = self.env['decimal.precision'].precision_get('Product Unit of Measure')
-            lines = self.filtered(
-                lambda r: r.state == 'sale' and not r.is_expense and float_compare(r.product_uom_qty, values['product_uom_qty'], precision_digits=precision) == -1)
+            previous_product_uom_qty = {}
```

Comment:
Prefer filtered + dict comprehension (which is duplicated here also). It is more readable and efficient ( |= is O(n) )